web/packages/design/src/RadioButton/RadioButton.tsx

```diff
@@ -0,0 +1,253 @@
+            other: the actual radio button, which is a native input control, and
+            a state indicator. Note that we avoid the usual "label with content"
+            trick, because we want to be able to use this component both with
+            and without surrounding labels. Instead, we use absolute positioning
+            and an actually rendered input with a custom appearance. */}
+          <RadioButtonInternal ref={ref} rbSize={size} {...inputProps} />
+          <Indicator rbSize={size} />
+        </InnerWrapper>
+      </OuterWrapper>
+    );
+  }
+);
+
+const OuterWrapper = styled.span`
+  display: inline-block;
+  line-height: 0;
+  margin: 3px;
+`;
+
+const InnerWrapper = styled.span`
+  display: inline-block;
+  position: relative;
+`;
+
+const Indicator = styled.span<{ rbSize?: RadioButtonSize }>`
+  border-radius: 50%;
+  position: absolute;
+  ${indicatorSize}
+  left: 0;
+  right: 0;
+  top: 0;
+  bottom: 0;
+  margin: auto;
+  pointer-events: none;
+  opacity: 0;
+
+  transition: all 150ms;
+
+  input:checked + & {
+    opacity: 1;
+  }
+
+  input:enabled + & {
+    background: ${props =>
+      props.theme.colors.interactive.solid.primary.default.background};
+  }
+
+  input:enabled:hover + &,
+  .teleport-radio-button__force-hover input + & {
+    background-color: ${props =>
+      props.theme.colors.interactive.solid.primary.hover.background};
+  }
+
+  input:enabled:focus-visible + &,
+  .teleport-radio-button__force-focus-visible input + & {
+    background-color: ${props =>
+      props.theme.colors.interactive.solid.primary.default.background};
+  }
+
+  input:enabled:active + &,
+  .teleport-radio-button__force-active input + & {
+    background-color: ${props =>
+      props.theme.colors.interactive.solid.primary.active.background};
+  }
+
+  input:disabled + & {
+    background-color: ${props => props.theme.colors.text.disabled};
+  }
+`;
+
+function indicatorSize(props: { rbSize?: RadioButtonSize }) {
+  const { rbSize = 'large' } = props;
+  let s = '';
+  switch (rbSize) {
+    case 'large':
+      s = '10px';
+      break;
+    case 'small':
+      s = '8px';
+      break;
+    default:
+      rbSize satisfies never;
+  }
+  return { width: s, height: s };
+}
+
+export const RadioButtonInternal = styled.input.attrs({ type: 'radio' })<{
+  rbSize?: RadioButtonSize;
+}>`
+  appearance: none;
+  border-style: solid;
+  border-color: ${props => props.theme.colors.text.muted};
+  border-radius: 50%;
+  background-color: transparent;
+  cursor: ${props => (props.disabled ? 'not-allowed' : 'pointer')};
+
+  position: relative;
+  margin: 0;
+
+  // Give it some animation, but don't animate focus-related properties.
+  transition:
+    border-color 150ms,
+    background-color 150ms,
+    box-shadow 150ms;
+
+  // State-specific styles. Note: the "force" classes are required for
+  // Storybook, where we want to show all the states, even though we can't
+  // enforce them.
```

Comment:
I thought triggering events/simulating user actions was possible thru Storybook? Could be misremembering though.

Reply:
Nvm, they're within the same story, my bad 🙈